src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3140,6 +3143,94 @@ void addCloneMethod()
                 members.Add(new SynthesizedRecordClone(this, memberOffset: members.Count, diagnostics));
             }
 
+            MethodSymbol addPrintMembersMethod()
+            {
+                var targetMethod = new SignatureOnlyMethodSymbol(
+                    WellKnownMemberNames.PrintMembersMethodName,
+                    this,
+                    MethodKind.Ordinary,
+                    Cci.CallingConvention.HasThis,
+                    ImmutableArray<TypeParameterSymbol>.Empty,
+                    ImmutableArray.Create<ParameterSymbol>(new SignatureOnlyParameterSymbol(
+                        TypeWithAnnotations.Create(compilation.GetWellKnownType(WellKnownType.System_Text_StringBuilder)),
+                        ImmutableArray<CustomModifier>.Empty,
+                        isParams: false,
+                        RefKind.None)),
+                    RefKind.None,
+                    isInitOnly: false,
+                    returnType: TypeWithAnnotations.Create(compilation.GetSpecialType(SpecialType.System_Boolean)),
+                    refCustomModifiers: ImmutableArray<CustomModifier>.Empty,
+                    explicitInterfaceImplementations: ImmutableArray<MethodSymbol>.Empty);
+
+                MethodSymbol printMembersMethod;
+                if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingPrintMembersMethod))
```

Comment:
Consider the case where the type defines `PrintMembers` with the same signature but has a custom modifier on say the parameter. Will this call return that method or will it return `false`? Expectation is that it will return `false` here because it's a different signature but I wanted to verify.

Reply:
We're using signature comparison that ignores modifiers. There is likely a problem in modreq/ambiguity scenario. I'll add some tests and fix.

```

   public static readonly MemberSignatureComparer RecordAPISignatureComparer = new MemberSignatureComparer(
            considerName: true,
            considerExplicitlyImplementedInterfaces: true,
            considerReturnType: false,
            considerTypeConstraints: false,
            considerCallingConvention: false,
            considerRefKindDifferences: true,
            typeComparison: TypeCompareKind.AllIgnoreOptions);
````



---
In reply to: [468930000](https://github.com/dotnet/roslyn/pull/46616#discussion_r468930000) [](ancestors = 468930000)

Reply:
We're okay in source (can't define such ambiguous methods). `FindValidPrintMethod` handles the ambiguity scenario in PE. I'll add test.

---
In reply to: [469421081](https://github.com/dotnet/roslyn/pull/46616#discussion_r469421081) [](ancestors = 469421081,468930000)